dev-tools/mage/dockerbuilder.go

```diff
@@ -185,6 +204,21 @@ func (b *dockerBuilder) dockerBuild() (string, error) {
 	if repository := b.ExtraVars["repository"]; repository != "" {
 		tag = fmt.Sprintf("%s/%s", repository, tag)
 	}
+
+	platform := fmt.Sprintf("%s/%s", "linux", b.Arch)
+
+	if runtime.GOARCH != b.Arch { // we need a cross-platform build, check if buildx supports the requested platform
+		if b.buildxPlatforms == nil {
+			return "", fmt.Errorf("cross-platform docker build requested, but buildx is not available")
```

Comment:
Hm, good point, I hadn't realized the docker server could be on a different arch than the Go runtime. What's the output of `docker info` on your setup? If it reports `x86_64`, then I could use that instead.

Reply:
hmm I don't see anything x86_64 being reported but it could be easily me missing it 🙂 

```
$ docker info
Client: Docker Engine - Community
 Version:    27.4.0
 Context:    lima-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.19.2
    Path:     /opt/homebrew/lib/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  2.31.0
    Path:     /opt/homebrew/lib/docker/cli-plugins/docker-compose

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 4
 Server Version: 27.3.1
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 88bf19b2105c8b17560993bee28a01ddc2f97182
 runc version: v1.2.2-0-g7cb3632
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.8.0-49-generic
 Operating System: Ubuntu 24.04.1 LTS
 OSType: linux
 Architecture: aarch64
 CPUs: 12
 Total Memory: 9.689GiB
 Name: lima-linux
 ID: 785679d9-51ad-4d7b-a6a5-33e95506572d
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Username: pkoutsovasilis
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

Reply:
Are you sure this won't just work with your current setup then? According to docker documentation, `DOCKER_DEFAULT_PLATFORM` is equivalent to using the `--platform` command line argument. If that works, then why does buildx not report that as an available platform?

Reply:
yes I did run the commands and this failed with the error you return. And I can validate that although I do `export DOCKER_DEFAULT_PLATFORM=linux/amd64` this is what I get
```
$ docker buildx inspect
Name:          lima-linux
Driver:        docker
Last Activity: 2024-12-11 13:56:43 +0000 UTC

Nodes:
Name:             lima-linux
Endpoint:         lima-linux
Status:           running
BuildKit version: v0.16.0
Platforms:        linux/arm64
Labels:
 org.mobyproject.buildkit.worker.moby.host-gateway-ip: 172.17.0.1
```

Reply:
That says the platform is actually `linux/arm64`... So if you do the following on `main`:

```
export DOCKER_DEFAULT_PLATFORM=linux/amd64
SNAPSHOT=true PACKAGES=docker DOCKER_VARIANTS=basic PLATFORMS="linux/amd64" EXTERNAL=true mage package
```

You get a `linux/amd64` container image?

Reply:
running these commands not with the code of this PR I get the following
```
$ docker image inspect docker.elastic.co/beats/elastic-agent:9.0.0-SNAPSHOT
[
    {
        "Id": "sha256:088768712c3b225ae91544e032e5d6b402bd648b698af5127bb927b6badb4878",
        "RepoTags": [
            "docker.elastic.co/beats/elastic-agent:9.0.0-SNAPSHOT"
        ],
        "RepoDigests": [],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2024-12-12T13:47:01.385673709+02:00",
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "elastic-agent",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/usr/share/elastic-agent:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "BEAT_SETUID_AS=elastic-agent",
                "ELASTIC_CONTAINER=true",
                "GODEBUG=madvdontneed=1",
                "LIBBEAT_MONITORING_CGROUPS_HIERARCHY_OVERRIDE=/"
            ],
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "/usr/share/elastic-agent",
            "Entrypoint": [
                "/usr/bin/tini",
                "--",
                "/usr/local/bin/docker-entrypoint"
            ],
            "OnBuild": null,
            "Labels": {
                "description": "Elastic Agent - single, unified way to add monitoring for logs, metrics, and other types of data to a host.",
                "io.k8s.description": "Elastic Agent - single, unified way to add monitoring for logs, metrics, and other types of data to a host.",
                "io.k8s.display-name": "Elastic-Agent image",
                "license": "Elastic License",
                "name": "elastic-agent",
                "org.label-schema.build-date": "2024-12-12T11:38:27Z",
                "org.label-schema.license": "Elastic License",
                "org.label-schema.name": "elastic-agent",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.url": "https://www.elastic.co/beats/elastic-agent",
                "org.label-schema.vcs-ref": "1ad02e31eb3bf968187bd29788de95a9e0fc75a7",
                "org.label-schema.vcs-url": "github.com/elastic/elastic-agent",
                "org.label-schema.vendor": "Elastic",
                "org.label-schema.version": "9.0.0-SNAPSHOT",
                "org.opencontainers.image.authors": "infra@elastic.co",
                "org.opencontainers.image.created": "2024-12-12T11:38:27Z",
                "org.opencontainers.image.licenses": "Elastic License",
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.title": "Elastic-Agent",
                "org.opencontainers.image.vendor": "Elastic",
                "org.opencontainers.image.version": "24.04",
                "release": "1",
                "summary": "elastic-agent",
                "url": "https://www.elastic.co/beats/elastic-agent",
                "vendor": "Elastic",
                "version": "9.0.0-SNAPSHOT"
            }
        },
        "Architecture": "amd64",
        ...
```

and running from inside the container

```
root@ba5e7db42c5d:/usr/share/elastic-agent# file data/elastic-agent-1ad02e/elastic-agent
data/elastic-agent-1ad02e/elastic-agent: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=564f019591890ab07e6e380aa27b8622f4f584d5, for GNU/Linux 3.2.0, stripped
```